src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Win32.cs

```diff
@@ -158,7 +158,7 @@ private unsafe void Monitor(AsyncReadState state)
                 continueExecuting = Interop.Kernel32.ReadDirectoryChangesW(
                     state.DirectoryHandle,
                     state.Buffer, // the buffer is kept pinned for the duration of the sync and async operation by the PreAllocatedOverlapped
-                    _internalBufferSize,
+                    (uint)state.Buffer.Length,
```

Comment:
I assume the problem here was a concurrent change to the InternalBufferSize could lead to corruption?  Is there an issue for this?

Reply:
Correct, this is trying to fix a possible corruption bug; there's not a GitHub issue, though.